lib/model/store.dart

```diff
@@ -767,3 +773,145 @@ class UpdateMachine {
   @override
   String toString() => '${objectRuntimeType(this, 'UpdateMachine')}#${shortHash(this)}';
 }
+
+extension Max<T extends num> on Iterable<T> {
+  /// Finds the maximum number in an [Iterable].
+  ///
+  /// Returns null if the [Iterable] is empty.
+  T? max() => isNotEmpty ? reduce(math.max) : null;
+}
+
+/// A data structure to keep track of message ids.
+class IdTracker {
+  final Set<int> _ids = {};
+
+  int _maxId = -1;
+
+  void add(int id) {
+    _ids.add(id);
+
+    if (id > _maxId) {
+      _maxId = id;
+    }
+  }
+
+  // TODO: remove
+
+  int maxId() {
+    if (_maxId == -1) {
+      _maxId = _ids.max() ?? -1;
+    }
+    return _maxId;
+  }
+
+  @override
+  String toString() {
+    return _ids.toString();
+  }
+}
+
+/// A data structure to keep track of stream and topic messages.
+class RecentSenders {
```

Comment:
For adding new data structures on PerAccountStore — anything more complicated than a single Map, anything that calls for its own class to encapsulate operations on it — we put them in separate files to help keep down the size of `store.dart`. (Particularly because `store.dart` has a lot of stuff with a different flavor to it, like `GlobalStore`, `LiveGlobalStore`, and `UpdateMachine`.)

For this, `recent_senders.dart` would be a good name.